### PR TITLE
Type hints for s3 storage sdk

### PR DESCRIPTION
**Pull request by askmeegs**
Adds PEP 484 type hints for `storage/s3-sdk`

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by danielbankhead — APPROVED** [review]

File: storage/s3-sdk/list_gcs_buckets.py [hunk 1]
```diff
@@ -17,10 +17,10 @@
 import argparse
 
 # [START storage_s3_sdk_list_buckets]
-import boto3
+import boto3  # type: ignore
 
 
-def list_gcs_buckets(google_access_key_id, google_access_key_secret):
+def list_gcs_buckets(google_access_key_id: str, google_access_key_secret: str):
```
Comment (dandhlee) [inline storage/s3-sdk/list_gcs_buckets.py:23]:
```suggestion
def list_gcs_buckets(
    google_access_key_id: str,
    google_access_key_secret: str
) -> None:
```

File: storage/s3-sdk/list_gcs_objects.py [hunk 2]
```diff
@@ -17,10 +17,12 @@
...
 
 # [START storage_s3_sdk_list_objects]
-import boto3
+import boto3  # type: ignore
 
 
-def list_gcs_objects(google_access_key_id, google_access_key_secret, bucket_name):
+def list_gcs_objects(
+    google_access_key_id: str, google_access_key_secret: str, bucket_name: str
+):
```
Comment (dandhlee) [inline storage/s3-sdk/list_gcs_objects.py:25]:
```suggestion
) -> None:
```

**Review by dandhlee — APPROVED** [review]
LGTM for Python style.

Nit: don't forget return type hints. It'd be also good to add in docstrings like
```
def list_foo(a: int, b: str) -> int:
   """lists foo.
   
   more description
   
   Args: 
       a: some docstring for a
       b: some docstring for b
   
   Returns:
       some docstring for returning int and conditions
```
but not needed for this PR.

**Comment by kweinmeister** [comment]
Could you also include `"enforce_type_hints": True` in `noxfile_config.py` to test your changes, and enforce them going forward?

**Comment by askmeegs** [comment]
> LGTM for Python style.
> 
> Nit: don't forget return type hints. It'd be also good to add in docstrings like
> 
> ```
> def list_foo(a: int, b: str) -> int:
>    """lists foo.
>    
>    more description
>    
>    Args: 
>        a: some docstring for a
>        b: some docstring for b
>    
>    Returns:
>        some docstring for returning int and conditions
> ```
> 
> but not needed for this PR.

Yes, there's a separate bug I'm doing to return a list of strings for this one, instead of print. 

> LGTM for Python style.
> 
> Nit: don't forget return type hints. It'd be also good to add in docstrings like
> 
> ```
> def list_foo(a: int, b: str) -> int:
>    """lists foo.
>    
>    more description
>    
>    Args: 
>        a: some docstring for a
>        b: some docstring for b
>    
>    Returns:
>        some docstring for returning int and conditions
> ```
> 
> but not needed for this PR.

Yes I'm working on a separate bug for these same samples, to add docstrings and return types instead of print statements. PR coming.

File: storage/s3-sdk/noxfile_config.py [hunk 3]
```diff
@@ -35,13 +35,13 @@ def get_service_account_email():
 TEST_CONFIG_OVERRIDE = {
     # A dictionary you want to inject into your test. Don't put any
     # secrets here. These values will override predefined values.
-    'envs': {
-        'HMAC_KEY_TEST_SERVICE_ACCOUNT': get_service_account_email(),
+    "envs": {
```
Comment (dandhlee) [inline storage/s3-sdk/noxfile_config.py:38]:
```suggestion
    "enforce_type_hints": True,
    "envs": {
```

File: storage/s3-sdk/noxfile_config.py [hunk 4]
```diff
@@ -35,13 +35,13 @@ def get_service_account_email():
 TEST_CONFIG_OVERRIDE = {
     # A dictionary you want to inject into your test. Don't put any
```
Comment (dandhlee) [inline storage/s3-sdk/noxfile_config.py:36]: Sorry added this suggestion in the wrong section before, this should be better.
```suggestion
    "enforce_type_hints": True,
    # A dictionary you want to inject into your test. Don't put any
```

**Comment by engelke** [comment]
Please include the internal bug number for our tracking: b/280881588. Thanks!

**Review by engelke — APPROVED** [review]
LGTM

**Review by rsamborski — APPROVED** [review]
Approved after implementing comments required for Py3.7 and 3.8 tests to pass.

**Pull request merged**